app/src/db/migrations/20220627000000_002-metadata-tags.js

```diff
@@ -0,0 +1,122 @@
+const stamps = require('../stamps');
+
+exports.up = function (knex) {
+  return Promise.resolve()
+
+    // // create metadata table
+    .then(() => knex.schema.createTable('metadata', table => {
+      table.specificType(
+        'id',
+        'integer GENERATED ALWAYS AS IDENTITY PRIMARY KEY'
+      );
+      table.string('key').index();
+      table.string('value').index();
+      table.unique(['key', 'value']);
+
+    }))
+
+    // create version_metadata table
+    .then(() => knex.schema.createTable('version_metadata', table => {
+      table.uuid('versionId').notNullable().references('id').inTable('version');
+      table.integer('metadataId').notNullable().references('id').inTable('metadata');
+      stamps(knex, table);
+      table.primary(['versionId', 'metadataId']);
+    }))
+
+    // create tag table
+    .then(() => knex.schema.createTable('tag', table => {
+      table.specificType(
+        'id',
+        'integer GENERATED ALWAYS AS IDENTITY PRIMARY KEY'
+      );
+      table.string('key').index();
+      table.string('value').index();
+      table.unique(['key', 'value']);
+    }))
+
+    // create version_tag table
+    .then(() => knex.schema.createTable('version_tag', table => {
+      table.uuid('versionId').notNullable().references('id').inTable('version');
+      table.integer('tagId').notNullable().references('id').inTable('tag');
+      stamps(knex, table);
+      table.primary(['versionId', 'tagId']);
+    }))
+
+
+    // Create metadata audit trigger
+    .then(() => knex.schema.raw(`CREATE TRIGGER audit_version_metadata_trigger
+    AFTER UPDATE OR DELETE ON version_metadata
+    FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();`))
+    // Create tag audit trigger
+    .then(() => knex.schema.raw(`CREATE TRIGGER audit_version_tag_trigger
+    AFTER UPDATE OR DELETE ON version_tag
+    FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();`))
```

Comment:
While technically we aren't updating the explicit metadata/tag table records themselves, they do still need to be audited. Please add those triggers in.

Reply:
the `if_modified_func()` that is triggered inserts the updated record (as JSON) into the audit log. 
it also looks for the 'updatedBy' field in the changed row to put in the audit record 'updatedByUsername' column.
..so the table being audited needs to have an updatedBy column. it can have a NULL value but must exist.

to audit this table i believe we would either need to modify that function or have the updatedBy/At columns.
probably something we should defer until we have a feature in coms where people can actually update/delete the tags themselves